### PR TITLE
procedures: Setting up GitLab OAuth and Che with DevWorkspace engine enabled

### PR DESCRIPTION
**Pull request by xbaran4**
Signed-off-by: Pavol Baran <pbaran@redhat.com>

## What does this pull request change?
Describes steps to authorize Che in GitLab with OAuth, when Che is deployed with DevWorkspaces engine on. (So there is no Keycloak)

## What issues does this pull request fix or reference?
https://github.com/eclipse-che/che-server/pull/195

## Specify the version of the product this pull request applies to
next

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

**Comment by xbaran4** [comment]
There is a PR check failing for reason outside the scope of this PR's content. Should I fix it or leave it as is?

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc [hunk 1]
```diff
@@ -14,7 +14,51 @@ OAuth2 for GitLab allows accepting factories from private GitLab repositories.
 
 .Procedure
 
-* Create a link:https://docs.gitlab.com/ee/integration/oauth_provider.html#authorized-applications[Authorized OAuth2 application in GitLab] using {prod-short} as the application `Name` and {identity-provider} GitLab endpoint URL as the value for `Redirect URI`. The callback URL default value is `++https://++keycloak-{prod-namespace}.__<DOMAIN>__/auth/realms/{prod-deployment}/broker/gitlab/endpoint`, where `__<DOMAIN>__` is {orch-name} cluster domain. Store the `Application ID` and `Secret` values. 
+* If the {devworkspace} engine is enabled, follow these steps:
```
Comment (themr0c) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc]: A `If` statement is a definitive sign that we should rather have 2 procedures and an assembly binding them.

* assembly_configuring-gitlab-oauth2.adoc 
* proc_configuring-gitlab-oauth2-with-the-che-server-engine.adoc 
* proc_configuring-gitlab-oauth2-with-devworkspace.adoc 

Are you feeling confident enough to do the reorganization, or do you need assistance from someone in the Docs Team?

Comment (xbaran4) [reply modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc]: I will try to reorganize it myself. I'll let you know if I get stuck at something.

Comment (xbaran4) [reply modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc]: I split it into two seperate procedures but wanted to keep it in the configuring authorization assembly. Let me know what you think.

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+type: Opaque
+data:
+  id: <...> <3>
+  secret: <...> <4>
+EOF
+----
+<1> {prod-short} namespace. The default is {prod-namespace}
+<2> URL of your GitLab service
+<3> base64 encoded GitLab `Application ID`
+<4> base64 encoded GitLab `Secret`
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]: @xbaran4, what about adding a link to the GitLab page:

```suggestion

.Additional resources
* link:https://docs.gitlab.com/ee/integration/oauth_provider.html[Configure GitLab as an OAuth 2.0 authentication identity provider]
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
+
+[id="configuring-gitlab-oauth2-with-devworkspace-engine_{context}"]
+= Configuring GitLab OAuth2 with {devworkspace} engine
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
= Configuring GitLab OAuth2 with the {devworkspace} engine
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+.Prerequisites
+
+* GitLab server is running and available from {prod-short}
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
* The GitLab server is running and available from {prod-short}.
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-che-server-engine.adoc [hunk 5]
```diff
@@ -13,8 +13,7 @@ OAuth2 for GitLab allows accepting factories from private GitLab repositories.
 * GitLab server is running and available from {prod-short}
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-che-server-engine.adoc]:
```suggestion
* The GitLab server is running and available from {prod-short}.
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+.Procedure
+
+. Create an OAuth2 link:https://docs.gitlab.com/ee/integration/oauth_provider.html#authorized-applications[Authorized application] in Gitlab:
+
+* Enter Che as the application `Name`
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
* Enter Che as the application `Name`.
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+* Enter the {prod-short} OAuth callback URL as `Redirect URI`. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback`.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
* Enter the {prod-short} OAuth callback URL as the `Redirect URI`. The default value is `++https://++{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback`.
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+* Check the `Confidential` and `Expire access tokens` checkboxes.
+
+* Check `api`, `write_repository`, `openid` scopes.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
* Under *Scopes*, check the `api`, `write_repository`, and `openid` checkboxes.
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    che.eclipse.org/scm-server-endpoint: <...> <2>
+type: Opaque
+data:
+  id: <...> <3>
+  secret: <...> <4>
+EOF
+----
+<1> {prod-short} namespace. The default is {prod-namespace}
+<2> URL of your GitLab service
+<3> base64 encoded GitLab `Application ID`
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc:47]:
```suggestion
<3> Base64 encoded GitLab `Application ID`
```

Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc:47]:
```suggestion
<3> Base64 encoded GitLab `Application ID`
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+. Create a new secret in the {orch-namespace} where {prod-short} is deployed.
++
+[subs="+quotes,+attributes"]
+----
+$ {orch-cli} apply -f - <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: gitlab-oauth-config
+  namespace: <...> <1>
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
  namespace: __<namespace>__ <1>
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  labels:
+    app.kubernetes.io/part-of: che.eclipse.org
+    app.kubernetes.io/component: oauth-scm-configuration
+  annotations:
+    che.eclipse.org/oauth-scm-server: gitlab
+    che.eclipse.org/scm-server-endpoint: <...> <2>
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
    che.eclipse.org/scm-server-endpoint: __<your_GitLab_service_URL>__
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+type: Opaque
+data:
+  id: <...> <3>
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
  id: __<base64_encoded_GitLab_Application_ID>__
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+  secret: <...> <4>
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
  secret: __<base64_encoded_GitLab_Secret>__
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc (hunk 2 above)
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc:48]:
```suggestion
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc (hunk 9 above)
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-che-server-engine.adoc [hunk 14]
```diff
@@ -3,8 +3,8 @@
 // Configuring GitLab OAuth2
 
 
-[id="configuring-gitlab-oauth2_{context}"]
-= Configuring GitLab OAuth2
+[id="configuring-gitlab-oauth2-with-che-server-engine_{context}"]
+= Configuring GitLab OAuth2 with {prod-deployment}-server engine
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-che-server-engine.adoc]:
```suggestion
= Configuring GitLab OAuth2 with the {prod-deployment}-server engine
```

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc (hunk 1 above)
Comment (max-cx) [reply modules/administration-guide/partials/proc_configuring-gitlab-oauth2.adoc]: @themr0c, looks like we're still missing the assembly that you suggested, but I looked at the build preview and it looks okay. I'm adopting proc_configuring-gitlab-oauth2-with-devworkspace.adoc for main anyway (with some editing). So, if this is okay with you, we can go ahead on this PR after my suggested changes are fixed.

**Review by max-cx — CHANGES_REQUESTED** [review]
@xbaran4, I added several suggested changes. Let me know when you've had a bit of time to review and hopefully apply these changes. When you let me know, I can approve the language part.

File: modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+EOF
+----
+<1> {prod-short} namespace. The default is {prod-namespace}
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_configuring-gitlab-oauth2-with-devworkspace-engine.adoc]:
```suggestion
<1> The {prod-short} namespace. The default is {prod-namespace}.
```

**Review by themr0c — APPROVED** [review]

**Comment by max-cx** [comment]
Migrating the content of this PR to the new docs/main in https://github.com/eclipse-che/che-docs/pull/2218

**Pull request merged**